django/db/models/fields/__init__.py

```diff
@@ -828,18 +828,17 @@ def get_flatchoices(self, include_blank=True,
         first_choice = blank_choice if include_blank else []
         return first_choice + list(self.flatchoices)
 
+    @warn_about_renamed_method('Field', '_get_val_from_obj', 'value_from_object',
+                               RemovedInDjango21Warning)
     def _get_val_from_obj(self, obj):
-        if obj is not None:
```

Comment:
Sorry, I am not perfect, and it took some time to understand the issue on my part. Django is sensitive about backwards compatibility so this is why I raised the issue. I doubt if the case in the test I offered is a common one such that it would prohibit the deprecation, but I wanted to offer some evidence for my comment about the implementations being slightly different. I don't see a problem in proceeding with the deprecation, but there needs to be some nuance in recommending `value_from_object()` as an identical replacement which isn't true, and we shouldn't needlessly change the implementation of `_get_val_from_obj()` while it's being deprecated such that code that's using it might break.